client/shared/src/main/scala/org/http4s/client/Client.scala

```diff
@@ -296,16 +297,15 @@ object Client {
         disposed: Ref[F, Boolean],
     ): Resource[F, Response[F]] =
       for {
-        channel <- Resource.make(
-          Channel.synchronous[F, Chunk[Byte]]
-        )(_.stream.compile.drain)
+        channel <- Resource.eval(Channel.synchronous[F, Chunk[Byte]])
 
-        _ <- response.body.chunks
+        producer = response.body.chunks
           .through(channel.sendAll)
           .compile
           .drain
           .uncancelable
-          .background
+
+        _ <- Resource.make(producer.start)(_ => channel.stream.compile.drain).map(_.join)
```

Comment:
We are discarding the result anyway, so no point to `.map`. And I think it's good practice to `join` the fiber after `draining` just to be sure that it completed.

```suggestion
        _ <- Resource.make(producer.start)(f => channel.stream.compile.drain *> f.join.void)
```